subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -29,6 +29,13 @@ LOG_MODULE_REGISTER(net_wifi_shell, LOG_LEVEL_INF);
 #include <zephyr/sys/slist.h>
 
 #include "net_shell_private.h"
+#ifdef CONFIG_WIFI_NM_WPA_SUPPLICANT_CRYPTO_ENTERPRISE
+#include "ca-cert.h"
+#include "client-cert.h"
+#include "client-key.h"
+#include "client-cert.h"
+#include "client-key.h"
```

Comment:
Ah, you want to use the certs here in the shell. I think user should be able to manipulate the certs at runtime, there is tls-certificate shell for that, we can use it to let the user to manipulate the certificates.

Reply:
Hi @jukkar  this is @krish2718 's suggestions that I should add the shell cmd to set the enterprise cert/key for wifi, you mean I should remove the shell cmd of enterprise_creds here?

Reply:
>  there is tls-certificate shell for that, we can use it to let the user to manipulate the certificates.

I had tried that, it's incredibly difficult to use, need to have custom configs to enable larger buffers and default SHELL RX ring is small, so, need to load `cred buf XXXX` multiple times at least 10 for a normal cert, we can certainly add this in the future, but just copying pem/header and rebuilding would be easy esp. as certs don't change often.

Reply:
Yes, the idea is to overwrite the cert/C header in `test_certs` and then rebuild, I can push a change that automates cert->C header generation instead of user running manual commands.

Reply:
Even if we store the certs in samples or somewhere else, the bigger question is how to manage them. I mean we have the TLS credentials API for storing the certificates, so passing them around like this in this PR looks a bit problematic. If using the TLS cred API the certs could be stored in secure storage instead of RAM / flash which is not secure at all.

Reply:
I took at a stab auto-generating these cert headers: https://github.com/krish2718/zephyr/commit/7254cae3f28f861bec7d06d5f192e60b105daad9

Reply:
> If using the TLS cred API the certs could be stored in secure storage instead of RAM / flash which is not secure at all.

This is purely for testing and not for production.

Reply:
> Zephyr has a certificate storage support, see tls_credential API https://docs.zephyrproject.org/latest/doxygen/html/group__tls__credentials.html. So instead of directly passing certs to around, couldn't we utilize the cert API for this?

Yes, we can use that, but for non-secure storage, we need to write a sample and then call `tls_credential_add` or use TLS `cred` shell, while in the sample we can load certs at compile time, I was trying to get `shell` working at compile time as shell at runtime is too much trouble.

Reply:
> I took at a stab auto-generating these cert headers: [krish2718@7254cae](https://github.com/krish2718/zephyr/commit/7254cae3f28f861bec7d06d5f192e60b105daad9)

Looks nice. Do you know if openssl always available in the build environment, for example in windows?

Reply:
> This is purely for testing and not for production.

Understood. I just want to make it possible so that the user is able to change the certs "easily" if they are stored next to wifi-shell. Which means either pem or der format files.

Reply:
The convertion to DEF was done so that we can use `xxd` to generate a C header, MbedTLS PEM/DER is independent to this.

Reply:
Hi @krish2718  thanks for your commit

> I took at a stab auto-generating these cert headers: [krish2718@7254cae](https://github.com/krish2718/zephyr/commit/7254cae3f28f861bec7d06d5f192e60b105daad9)

Reply:
Hi @jukkar @krish2718 can this PR be approved, or what should I continue to change?

Reply:
>> I took at a stab auto-generating these cert headers: [krish2718@7254cae](https://github.com/krish2718/zephyr/commit/7254cae3f28f861bec7d06d5f192e60b105daad9)

> Looks nice. Do you know if openssl always available in the build environment, for example in windows?

Yes, OpenSSL is available for Windows and Mac.

Reply:
As the certs are only used for testing, and it seems to be quite difficult to solve this otherwise, and to get this PR forward, I am reluctantly inclined to accept this. But I want the cert merging to be easier for the end user, there is no need for the user to create the header file as that can be done automatically (if using der files) using `generate_inc_file_for_target` CMake macro.

Reply:
> Hi @jukkar @krish2718 can this PR be approved, or what should I continue to change?

I will share a commit using `generate_inc_file_for_target` and some re-org and then you can squash that in to your commit in this PR.

Reply:
> generate_inc_file_for_target

This was a great suggestion, thanks. It actually can take PEM and convert C header (used in Zephyr already), so, did the following changes: https://github.com/krish2718/zephyr/commit/bd3ecb62cccc3a12b3b0d245a582281cce807951 

1. Moved certs to `samples/net/wifi/test_certs`
2. Modified `wifi_shell.c` and corresponding CMake file to auto-generate the headers.

This approach looks much better.

@MaochenWang1  Please cherry-pick my commit and `fixup` with original one, I have only build-tested this, so, please test to make sure that certs are actually working.

Reply:
Hi @krish2718 thanks for your help, I cherry-pick your commit, and I use der to replace pem, as I found generate_inc_file_for_target only works for der to .h, please review @jukkar

Reply:
there are compliance check warnings that I don't know how to fix:
![image](https://github.com/user-attachments/assets/52f16fe1-a896-4111-89c6-ddc76f1a5ea7)
but I see in /zephyr/samples/net/sockets/echo_client/src, there is echo-apps-cert.der

Reply:
I was finally able to get WPA2-EAP-TLS working on nRF7002DK, see my branch for latest fixes: https://github.com/krish2718/zephyr/tree/main_zephyr_10 I have squashed the commit, so, you can just cherry-pick and rebase and it's good to go. Apart from above, mainly added

* Heap adjustments
* SSL content length fix
* Add missing MbedTLS configs

I am working on adding individual configurations for Entperprise methods e.g., choose EAP-TLS only or EAP-PEAP only and we also need to move the respective MbedTLS dependencies to the individual methods, will submit a new PR for that.

Reply:
Also, https://github.com/krish2718/sdk-hostap/commit/61e37b46571ef25efdaef7744e46c5033ff30bab this fix is needed, I am not sure if this is applicable for all Enterprise methods or just EAP-TLS, I am sure there is a proper way to fix this, but MbedTLS itself hardcoded the length to `48` in the callback invocation see https://github.com/Mbed-TLS/mbedtls/blob/195e1647b24d231521675c0aab6d4715bb579be5/library/ssl_tls.c#L9114

Reply:
Hi @jukkar  @krish2718  please review again.